### PR TITLE
BTFS-1651

### PR DESCRIPTION
**Pull request by steveyeom**

### CODE REVIEW

**Comment by steveyeom** [comment]
Test OK with TestStorageUploadWithOnSign/OffSign in NON-custom mode.

File: storage.go [hunk 1]
```diff
@@ -166,11 +168,61 @@ func (s *Shell) GetUts() string {
...
-	//offline session signature
-	now := time.Now()
-	sessionSignature := fmt.Sprintf("%s:%s:%s", utils.GetPeerId(), hash, "time.Now().String()")
-	return sessionSignature, now, nil
+func NewSessionSignature(hash string, peerIdStr string, uts string, verifyBefore bool) (string, error) {
+	// Create offline session signature input data
+	inputDataStr := fmt.Sprintf("%s%s%s", hash, peerIdStr, uts)
+
+	// Sign sessionSignature
+	privKey, _ := crypto.ToPrivKey(utils.GetPrivateKey())
```
Comment (taiyangc) [inline storage.go]: check err

File: storage.go [hunk 2]
```diff
@@ -166,11 +168,61 @@ func (s *Shell) GetUts() string {
...
+
+	peerId, err := peer.IDB58Decode(utils.GetPeerId())
+	if err != nil {
+		return "", err
+	}
+
+	if verifyBefore {
+		err = VerifySessionSignature(peerId, inputDataStr, sigStr)
+		if err != nil {
+			return "", errors.New("session signature verification failed")
```
Comment (taiyangc) [inline storage.go]: why not return `err` directly?

**Review by taiyangc — COMMENTED** [review]
small comments

**Review by taiyangc — APPROVED** [review]

**Pull request merged**